core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2432,6 +2432,30 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[util] object CallerContext extends Logging {
+  private val callerContextClassExists = {
+    try {
+      // scalastyle:off classforname
+      Class.forName("org.apache.hadoop.ipc.CallerContext")
+      Class.forName("org.apache.hadoop.ipc.CallerContext$Builder")
+      // scalastyle:on classforname
+      true
+    } catch {
+      case _: ClassNotFoundException =>
+        false
+      case NonFatal(e) =>
+        logWarning("Fail to load the CallerContext class", e)
+        false
+    }
+  }
+
+  private val conf = SparkHadoopUtil.get.conf
```

Comment:
This doesn't need to be a separate member; it can be a local val.
Still, why have more than one member here? Initializing `callerContextSupported` in one block of code is sufficient and simpler.